File: ch04/23/PrintFromTopToBottom.cpp
/*
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};*/
class Solution {
public:
    
    //第二遍刷题
    vector<int> PrintFromTopToBottom(TreeNode* root) {  
        
        vector<int> reVec; //初始一个空向量用于返回
        queue<TreeNode*> tempQueue;//辅助队列
        
        if (root == NULL)
            return reVec;
        
        tempQueue.push(root);
        while (!tempQueue.empty()) {
            TreeNode * tmpNode = tempQueue.front();
            reVec.push_back(tmpNode->val);
            if (tmpNode->left) 
                tempQueue.push(tmpNode->left);
            if (tmpNode->right)
                tempQueue.push(tmpNode->right);
            
          	tempQueue.pop();
        }
        
        return reVec;
    }
};

File: ch04/23/PrintFromTopToBottom.py
# 从上往下打印出二叉树的每个节点，同层节点从左至右打印
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回从上到下每个节点值列表，例：[1,2,3]
    def PrintFromTopToBottom(self, root):
        # write code here
        re_lst = [] 
        queue = [] # 用列表作为队列
        if root == None:
            return re_lst

        queue.append(root)
        while len(queue) != 0:
            tmp = queue[0]
            if tmp.left:
                queue.append(tmp.left)
            if tmp.right:
                queue.append(tmp.right)
            re_lst.append(tmp.val)
            queue.pop(0)
        
        return re_lst

